packages/frontend/src/app/events/page.tsx

```diff
@@ -0,0 +1,75 @@
+'use client';
+
+import { useSearchParams } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+
+import PaginationControls from '@/components/atoms/PaginationControls';
+import EventsDisplay from '@/components/molecules/EventsDisplay';
+import EventSearchPanel from '@/components/molecules/EventSearchPanel';
+import { EventService } from '@/services/EventService';
+import { Event } from '@/types';
+
+const EventPage = () => {
+  const searchParams = useSearchParams();
+  const [events, setEvents] = useState<Event[]>([]);
+  const [totalItems, setTotalItems] = useState<number>(0);
+  const [error, setError] = useState<string | null>(null);
+
+  const [title, setTitle] = useState<string>('');
+  const [tags, setTags] = useState<string[]>([]);
+  const [type, setType] = useState<string>('');
+  const [page, setPage] = useState<number>(searchParams.get('page') ? Number(searchParams.get('page')) : 1);
+
+  const per_page = searchParams.get('per_page') ? Number(searchParams.get('per_page')) : 12;
+
+  useEffect(() => {
+    try {
+      new EventService().getEvents(type, tags, title, page, per_page).then(({ events }) => {
```

Comment:
A common problem with this approach will be caching the fetched data. I would use a react hooks library instead of calling the event Service, the library will handle cache invalidation, updates, and error handling which frees us from repeating the code in multiple modules. Let's use https://swr.vercel.app/docs/getting-started, its made by the same team behind Next.js